### DISCUSSION TITLE
Easily merge pair_lists

### DISCUSSION THREAD

**Discussion by felixbjorkeson** [openMVG/openMVG#2108] [Q&A] [answered]
Is it safe to append one pair_lists to another (and in the process creating duplicates)?
I want to combine video and GPS matching pairs. The data I am using is sequential but there are multiple overlaps (where the drone turns around and sees the same area again)
Using only GPS fails quite often to create pairs for neighboring images which should match. I can increase the --neighbor_count to get a bit better results but then other images get very many pairs.

**Comment by pmoulon** [comment, marked as answer]
Yes, you can merge pair_lists, duplicates will be ignored.

Please let us know how it goes.

I was wondering about adding a mode where we could add to existing pair_lists within the binary, we can think about that depending of the results of your experiments ;-)

**Comment by felixbjorkeson** [comment]
Yes the merge works, which is nice. Having a mode to do it automatically would be nicer :P
Another thing I reacted on is that the GPS mode should have some distance threshold either together with or instead of the N neighbors.
It is quite difficult getting a good pair graph when the data set is from a drone, looking straight down flying in a back and forth pattern with some side overlap. Video mode will only get pairs in the same "lane". Gps mode need very high N (so that some views from neighboring lanes are included) and if the density in the same lane decreases, views that are too far away will suddenly be included.

**Reply by pmoulon** [reply]
Thank you for your feedback. One last piece of information. Indeed heuristics on video mode and gps position cannot solve it all perfectly since we don't know the orientation of the camera frustum. But one way to overcome this is to mix image base similarity pair list and heuristic pair list so at the end you would have best candidate from image similarity and proximity. 

For the image similarity Openmvg provides 3 way:
- full matching (fundamental, essential or fast Ortho essential)
- approximated matching (pre emptive matching -> using only the largest scale features to make search fast)
- approximated matching (Vlad - compute a global embedding for the image descriptor and find the closest image with that)

It could be nice to work together on a wiki tutorial about available image matching method.
